topics/go/algorithms/data/hash/hash.go

```diff
@@ -0,0 +1,107 @@
+// Package hash implements a hash table
+package hash
+
+import (
+	"fmt"
+	"hash/maphash"
+)
+
+const (
+	numBuckets = 256 // Number of buckets in hash
+)
+
+// entry in the hash table
+type entry struct {
+	key   string
+	value int
+}
+
+// Hash is a simple Hash table
+type Hash struct {
+	buckets [][]*entry
+	h       maphash.Hash
+}
+
+// NewHash returns a new hash table
+func NewHash() *Hash {
+	return &Hash{
+		buckets: make([][]*entry, numBuckets),
+	}
+}
+
+// Set sets the key to value in the hash
+func (h *Hash) Set(key string, value int) {
+	b := h.hashKey(key)
```

Comment:
It seems this hashkey guarentees a value inside the length of the slice?

Reply:
Yup

Reply:
Can we explain how that works?